packages/p2p/src/utils/orders.js

```diff
@@ -0,0 +1,152 @@
+    const is_completed_order = status === 'completed';
+    const is_expired_order = status === 'timed-out';
+    const is_pending_order = status === 'pending';
+    const is_refunded_order = status === 'refunded';
+
+    const is_my_ad = loginid === advertiser_details.loginid;
+
+    const is_active_order = is_pending_order || is_buyer_confirmed_order || is_expired_order;
+    const is_inactive_order = is_buyer_cancelled_order || is_refunded_order || is_completed_order;
+
+    // Status highlighting (alert = yellow, danger = red)
+    const should_highlight_alert = (is_buy_ad && is_buyer_confirmed_order) || (!is_buy_ad && is_pending_order);
+    const should_highlight_danger = (is_buy_ad && is_pending_order) || (!is_buy_ad && is_buyer_confirmed_order);
+
+    // Order details footer.
+    let should_show_cancel_and_paid_button, should_show_complain_and_received_button, status_string;
+
+    if (is_buy_ad) {
+        should_show_cancel_and_paid_button = !is_my_ad && is_pending_order;
+        should_show_complain_and_received_button = is_my_ad && (is_buyer_confirmed_order || is_expired_order);
+    } else {
+        should_show_cancel_and_paid_button = is_my_ad && is_pending_order;
+        should_show_complain_and_received_button = !is_my_ad && (is_buyer_confirmed_order || is_expired_order);
+    }
+
+    const should_show_only_complain_button = is_expired_order;
+
+    const should_show_order_footer =
+        should_show_cancel_and_paid_button ||
+        should_show_complain_and_received_button ||
+        should_show_only_complain_button;
+
+    if (is_pending_order) {
+        if (is_my_ad) {
+            status_string = is_buy_ad ? localize('Wait for payment') : localize('Pay now');
+        } else {
+            status_string = is_buy_ad ? localize('Pay now') : localize('Wait for payment');
+        }
+    } else if (is_buyer_cancelled_order) {
+        status_string = localize('Cancelled');
+    } else if (is_buyer_confirmed_order) {
+        if (is_my_ad) {
+            status_string = is_buy_ad ? localize('Confirm payment') : localize('Wait for release');
+        } else {
+            status_string = is_buy_ad ? localize('Wait for release') : localize('Confirm payment');
+        }
+    } else if (is_expired_order || is_refunded_order) {
+        status_string = localize('Expired');
+    } else if (is_completed_order) {
+        status_string = localize('Completed');
+    } else {
+        status_string = localize('Unknown');
+    }
+
+    return {
+        ...order_details,
+        is_active_order,
+        is_buy_ad,
+        is_buyer_cancelled_order,
+        is_buyer_confirmed_order,
+        is_completed_order,
+        is_expired_order,
+        is_inactive_order,
+        is_my_ad,
+        is_pending_order,
+        is_refunded_order,
+        is_sell_ad,
+        labels: getOrderLabels(is_my_ad, is_buy_ad, order_details),
+        my_user_details: is_my_ad ? advertiser_details : client_details,
```

Comment:
can we make this into variable instead of dirtying return object

Reply:
applied to until line 143

Reply:
Why? We're not re-using any of those variables other than returning them as object values.

Reply:
yes, you're right, that is fine too